### PR TITLE
refactor: replace reflect in userFromInnerEvent

### LINKED ISSUE #140: refactor(core): replace reflection in userFromInnerEvent with type-safe switch
## Summary

The `userFromInnerEvent` function in `core/events_helper.go` uses runtime reflection to extract the `User` field from Slack event structs:

```go
func userFromInnerEvent(event *slackevents.EventsAPIInnerEvent) string {
	return reflect.ValueOf(event.Data).Elem().FieldByName("User").String()
}
```

This approach has several risks:

1. **Silent failures** — if `slack-go/slack` renames the `User` field in a future version (which is plausible given the 9-version gap tracked in #137), reflection will silently return an empty string instead of a compile error. This would cause the bot to silently drop events or misattribute users.
2. **No type safety** — there's no compile-time guarantee that every event type handled in the switch statement in `core/core.go` has a `User` field. If a new event type is added, the reflection call may panic or return empty.
3. **Performance overhead** — reflection on every incoming event is unnecessary when the supported event types are known and finite (currently `AppMentionEvent` and `MessageEvent`).

## Proposed Solution

Replace the reflection-based approach with an explicit type switch:

```go
func userFromInnerEvent(event *slackevents.EventsAPIInnerEvent) string {
	switch ev := event.Data.(type) {
	case *slackevents.AppMentionEvent:
		return ev.User
	case *slackevents.MessageEvent:
		return ev.User
	default:
		return ""
	}
}
```

This mirrors the same type switch already used in `handleEvent()` in `core/core.go:411` and ensures compile-time type checking. It also makes adding new event types a conscious decision rather than an implicit reflection lookup.

## Acceptance Criteria

- [ ] Replace reflection with explicit type switch in `core/events_helper.go`
- [ ] Remove the `reflect` import
- [ ] All existing tests pass
- [ ] Add a test case for unsupported event types returning empty string

## Related

- #137 — slack-go/slack dependency update (may change field names, making this fix more urgent)

### PR DESCRIPTION
**Pull request by iamAdityafr**
## Changes

- Replace reflection with explicit type switch in `core/events_helper.go`
- Remove the reflect import
- Add a test cases for unsupported event types returning empty string

Fixes #140

### CODE REVIEW

**Review by jgnagy — APPROVED** [review]
## Review by Hermes Agent ✅

**Verdict: Approved** — clean, well-scoped refactor that does exactly what issue #140 asked.

### What this does
Replaces the reflection-based `userFromInnerEvent` with an explicit type switch over the two event types the codebase actually handles (`AppMentionEvent` and `MessageEvent`). Removes the `reflect` import.

### Correctness
The original `reflect.ValueOf(event.Data).Elem().FieldByName("User").String()` would silently extract a `User` field from *any* slackevents type that happened to have one — there are ~19 such types in slack-go v0.18.0. The new code returns `""` for all types except the two handled in `core.go`'s dispatch switch (lines 411–453), which are the only two that matter. For unhandled event types, returning `""` is actually *more* correct than silently pulling a User ID from an event type the application doesn't process.

### Tests
Good coverage — tests both supported event types, pointer vs. value receivers, a struct with no `User` field, and a bare `struct{}{}`. All pass, along with the full test suite.

### Minor non-blocking notes
- The test function name `TestUnsupportedEventType` slightly undersells its scope — it also tests the *supported* cases. `TestUserFromInnerEvent` would be more descriptive, but this is cosmetic.
- The new test uses raw `t.Errorf` while the rest of the file uses testify's `assert.Equal`. Consistency would be nice but isn't worth blocking on.

### Security
Removing reflection is a net positive — eliminates a class of silent runtime failures if slack-go changes field names.

Thanks for the contribution!

File: core/core_test.go [hunk 1]
```diff
@@ -74,3 +75,66 @@ func TestGlobalAdminsFromString(t *testing.T) {
 		})
 	}
 }
+
+func TestUnsupportedEventType(t *testing.T) {
```
Comment (jgnagy) [inline core/core_test.go]: 💡 **Non-blocking:** Consider renaming to `TestUserFromInnerEvent` — this tests both supported and unsupported event types, so the current name undersells the coverage.

File: core/core_test.go [hunk 2]
```diff
@@ -74,3 +75,66 @@ func TestGlobalAdminsFromString(t *testing.T) {
...
+				Data: struct{}{},
+			},
+			expected: "",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := userFromInnerEvent(tt.event)
+			if got != tt.expected {
```
Comment (jgnagy) [inline core/core_test.go]: 💡 **Non-blocking:** The rest of this file uses testify's `assert.Equal`. Consistency would be nice, but not worth a follow-up commit on its own.

**Review by jgnagy — DISMISSED** [review]
Thanks for this contribution! This is a clean refactor that replaces the reflection-based approach with a type-safe switch, exactly as described in issue #140.

**Verification:**
- ✅ All CI checks pass (lint, test on Go 1.25 & 1.26, build)
- ✅ Locally verified: `go test ./...` passes, `gofmt -l .` clean, `go vet ./...` clean
- ✅ Removes the `reflect` import as intended
- ✅ Test coverage added for supported event types (AppMentionEvent, MessageEvent) and unsupported types (ChannelCreatedEvent, FileDeletedEvent, empty struct)

**One minor observation:** The test function is named `TestUnsupportedEventType` but it also tests supported event types (AppMentionEvent returns "U123", MessageEvent returns "U456"). A name like `TestUserFromInnerEvent` would more accurately describe the full test scope. This is non-blocking — the tests themselves are correct and valuable.

This PR is ready for merge. Fixes #140.
